lib/std/io/buffered_tee.zig

```diff
@@ -0,0 +1,388 @@
+//! BufferedTee provides reader interface to the consumer. Data read by consumer
+//! is also written to the output. Output is hold lookahead_size bytes behind
+//! consumer. Allowing consumer to put back some bytes to be read again. On flush
+//! all consumed bytes are flushed to the output.
+//!
+//!       input   ->   tee   ->   consumer
+//!                     |
+//!                  output
+//!
+//! input - underlying unbuffered reader
+//! output - writer, receives data read by consumer
+//! consumer - uses provided reader interface
+//!
+//! If lookahead_size is zero output always has same bytes as consumer.
+//!
```

Comment:
The namespace of this file isn't accessible to users since `BufferedTee` and `bufferedTee` are exposed directly under `std.io` via `pub const BufferedTee = @import("io/buffered_tee.zig").BufferedTee;` in `io.zig`, so this doc comment won't show up in the generated documentation. If you move this right above `pub fn BufferedTee` and change it to a regular doc comment (`///`), then it'll show up as the documentation for that type.